chain/chain/src/resharding/manager.rs

```diff
@@ -54,15 +53,18 @@ impl ReshardingManager {
 
         let next_epoch_id = self.epoch_manager.get_next_epoch_id_from_prev_block(prev_hash)?;
         let next_shard_layout = self.epoch_manager.get_shard_layout(&next_epoch_id)?;
-        let children_shard_uids =
-            next_shard_layout.get_children_shards_uids(shard_uid.shard_id()).unwrap();
 
+        let resharding_event_type =
+            ReshardingEventType::from_shard_layout(&next_shard_layout, *block_hash, *prev_hash);
         // Hack to ensure this logic is not applied before ReshardingV3.
         // TODO(#12019): proper logic.
-        if next_shard_layout.version() < 3 || children_shard_uids.len() == 1 {
+        if next_shard_layout.version() < 3 {
             return Ok(());
         }
-        assert_eq!(children_shard_uids.len(), 2);
+        let Ok(Some(ReshardingEventType::SplitShard(split_shard_event))) = resharding_event_type
```

Comment:
You are not handling the error case here - Err and Ok(None) are treated the same. 
- if intentional - please comment
- if accidental - please add `?` or handle it explicitly